Core/src/org/sleuthkit/autopsy/casemodule/services/TagNameDefiniton.java

```diff
@@ -152,15 +164,37 @@ private String toSettingsFormat() {
         Set<TagNameDefiniton> tagNames = new HashSet<>();
         String setting = ModuleSettings.getConfigSetting(TAGS_SETTINGS_NAME, TAG_NAMES_SETTING_KEY);
         if (null != setting && !setting.isEmpty()) {
+            List<String> badTags = null;
             List<String> tagNameTuples = Arrays.asList(setting.split(";"));
             for (String tagNameTuple : tagNameTuples) {
                 String[] tagNameAttributes = tagNameTuple.split(",");
-                tagNames.add(new TagNameDefiniton(tagNameAttributes[0], tagNameAttributes[1], TagName.HTML_COLOR.valueOf(tagNameAttributes[2])));
+                if (tagNameAttributes.length == 3) {
+                    if (badTags == null) {
+                        String badTagsStr = ModuleSettings.getConfigSetting("CentralRepository", "db.badTags"); // NON-NLS
+                        if (badTagsStr == null) {
+                            badTagsStr = STANDARD_NOTABLE_TAG_DISPLAY_NAMES;
+                        }
+                        if (badTagsStr.isEmpty()) {
+                            badTags = new ArrayList<>();
+                        } else {
+                            badTags = new ArrayList<>(Arrays.asList(badTagsStr.split(",")));
+                        }
+                    }
+                    if (badTags.contains(tagNameAttributes[0])) {
+                        tagNames.add(new TagNameDefiniton(tagNameAttributes[0], tagNameAttributes[1], TagName.HTML_COLOR.valueOf(tagNameAttributes[2]), TskData.FileKnown.BAD));
+                    } else {
+                        tagNames.add(new TagNameDefiniton(tagNameAttributes[0], tagNameAttributes[1], TagName.HTML_COLOR.valueOf(tagNameAttributes[2]), TskData.FileKnown.UNKNOWN)); //add the default value for that tag 
+                    }
+                } else if (tagNameAttributes.length == 4) {
+                    tagNames.add(new TagNameDefiniton(tagNameAttributes[0], tagNameAttributes[1], TagName.HTML_COLOR.valueOf(tagNameAttributes[2]), TskData.FileKnown.valueOf(tagNameAttributes[3])));
+                }
+
```

Comment:
I do not understand what is going on in this code block. Let's discuss it, then comment it.